commit 45755006d88fdf60796dc0ca6611ebbd6227925c
Author: Susan Franklin <sfranklin>
Date:   Tue Mar 11 02:35:55 2008 +0000

    Bug 222156 - [ui] Can't add an artifact repository with the same URL than a metadata repo

1	0	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIMessages.java
26	2	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddArtifactRepositoryDialog.java
1	0	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/messages.properties
1	1	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/DefaultURLValidator.java
8	0	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/ProvisioningUtil.java
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIMessages.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIMessages.java
index 7b30d5c..d7cdcc6 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIMessages.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/ProvAdminUIMessages.java
@@ -24,6 +24,7 @@ public class ProvAdminUIMessages extends NLS {
 		// load message values from bundle file
 		NLS.initializeMessages(BUNDLE_NAME, ProvAdminUIMessages.class);
 	}
+	public static String AddArtifactRepositoryDialog_DuplicateURL;
 	public static String AddArtifactRepositoryDialog_OperationLabel;
 	public static String AddProfileDialog_Title;
 	public static String AddRepositoryDialog_InvalidURL;
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddArtifactRepositoryDialog.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddArtifactRepositoryDialog.java
index 8787e3f..ee1c521 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddArtifactRepositoryDialog.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddArtifactRepositoryDialog.java
@@ -11,10 +11,13 @@
 package org.eclipse.equinox.internal.p2.ui.admin.dialogs;
 
 import java.net.URL;
+import org.eclipse.core.runtime.*;
+import org.eclipse.equinox.internal.p2.ui.admin.ProvAdminUIActivator;
 import org.eclipse.equinox.internal.p2.ui.admin.ProvAdminUIMessages;
+import org.eclipse.equinox.internal.provisional.p2.core.ProvisionException;
 import org.eclipse.equinox.internal.provisional.p2.ui.dialogs.AddRepositoryDialog;
-import org.eclipse.equinox.internal.provisional.p2.ui.operations.AddArtifactRepositoryOperation;
-import org.eclipse.equinox.internal.provisional.p2.ui.operations.ProvisioningOperation;
+import org.eclipse.equinox.internal.provisional.p2.ui.dialogs.DefaultURLValidator;
+import org.eclipse.equinox.internal.provisional.p2.ui.operations.*;
 import org.eclipse.swt.widgets.Shell;
 
 /**
@@ -32,4 +35,25 @@ public class AddArtifactRepositoryDialog extends AddRepositoryDialog {
 	protected ProvisioningOperation getOperation(URL url) {
 		return new AddArtifactRepositoryOperation(ProvAdminUIMessages.AddArtifactRepositoryDialog_OperationLabel, url);
 	}
+
+	protected DefaultURLValidator createURLValidator() {
+		return new DefaultURLValidator() {
+			protected IStatus validateRepositoryURL(URL location, boolean contactRepositories, IProgressMonitor monitor) {
+				IStatus duplicateStatus = Status.OK_STATUS;
+				URL[] knownRepositories;
+				try {
+					knownRepositories = ProvisioningUtil.getArtifactRepositories(repoFlag);
+				} catch (ProvisionException e) {
+					knownRepositories = new URL[0];
+				}
+				for (int i = 0; i < knownRepositories.length; i++) {
+					if (knownRepositories[i].toExternalForm().equalsIgnoreCase(location.toExternalForm())) {
+						duplicateStatus = new Status(IStatus.ERROR, ProvAdminUIActivator.PLUGIN_ID, LOCAL_VALIDATION_ERROR, ProvAdminUIMessages.AddArtifactRepositoryDialog_DuplicateURL, null);
+						break;
+					}
+				}
+				return duplicateStatus;
+			}
+		};
+	}
 }
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/messages.properties b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/messages.properties
index 6790c6c..07184ab 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/messages.properties
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/messages.properties
@@ -25,6 +25,7 @@ AddRepositoryDialog_InvalidURL=Invalid URL
 AddProfileDialog_Title=Add Profile
 AddProfileDialog_OperationLabel=Add Profile
 AddMetadataRepositoryDialog_OperationLabel=Add Metadata Repository
+AddArtifactRepositoryDialog_DuplicateURL=Duplicate URL
 AddArtifactRepositoryDialog_OperationLabel=Add Artifact Repository
 MetadataRepositoriesView_AddRepositoryLabel=Add repository...
 MetadataRepositoriesView_AddRepositoryTooltip=Add a new metadata repository
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/DefaultURLValidator.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/DefaultURLValidator.java
index 59e58a4..42dcda6 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/DefaultURLValidator.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/DefaultURLValidator.java
@@ -25,7 +25,7 @@ import org.eclipse.equinox.internal.provisional.p2.ui.operations.ProvisioningUti
  */
 public class DefaultURLValidator extends URLValidator {
 
-	int repoFlag;
+	protected int repoFlag;
 
 	public DefaultURLValidator() {
 		repoFlag = IMetadataRepositoryManager.REPOSITORIES_ALL;
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/ProvisioningUtil.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/ProvisioningUtil.java
index d3da7c9..c771834 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/ProvisioningUtil.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/operations/ProvisioningUtil.java
@@ -169,6 +169,14 @@ public class ProvisioningUtil {
 		return manager.getKnownRepositories(flags);
 	}
 
+	public static URL[] getArtifactRepositories(int flags) throws ProvisionException {
+		IArtifactRepositoryManager manager = (IArtifactRepositoryManager) ServiceHelper.getService(ProvUIActivator.getContext(), IArtifactRepositoryManager.class.getName());
+		if (manager == null) {
+			throw new ProvisionException(ProvUIMessages.ProvisioningUtil_NoRepositoryManager);
+		}
+		return manager.getKnownRepositories(flags);
+	}
+
 	/*
 	 * Get the plan for the specified install operation
 	 */
commit 6783195773891c944684cccdaa0b03c2e3734df3
Author: Susan Franklin <sfranklin>
Date:   Tue Mar 11 02:52:13 2008 +0000

    Bug 222156 - [ui] Can't add an artifact repository with the same URL than a metadata repo

3	3	bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddArtifactRepositoryDialog.java
3	2	bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/AddColocatedRepositoryDialog.java
2	2	bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/externalFiles/MetadataGeneratingURLValidator.java
7	4	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/AddRepositoryDialog.java
62	0	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/DefaultMetadataURLValidator.java
0	62	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/DefaultURLValidator.java
diff --git a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddArtifactRepositoryDialog.java b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddArtifactRepositoryDialog.java
index ee1c521..dea2973 100644
--- a/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddArtifactRepositoryDialog.java
+++ b/bundles/org.eclipse.equinox.p2.ui.admin/src/org/eclipse/equinox/internal/p2/ui/admin/dialogs/AddArtifactRepositoryDialog.java
@@ -16,7 +16,7 @@ import org.eclipse.equinox.internal.p2.ui.admin.ProvAdminUIActivator;
 import org.eclipse.equinox.internal.p2.ui.admin.ProvAdminUIMessages;
 import org.eclipse.equinox.internal.provisional.p2.core.ProvisionException;
 import org.eclipse.equinox.internal.provisional.p2.ui.dialogs.AddRepositoryDialog;
-import org.eclipse.equinox.internal.provisional.p2.ui.dialogs.DefaultURLValidator;
+import org.eclipse.equinox.internal.provisional.p2.ui.dialogs.URLValidator;
 import org.eclipse.equinox.internal.provisional.p2.ui.operations.*;
 import org.eclipse.swt.widgets.Shell;
 
@@ -36,8 +36,8 @@ public class AddArtifactRepositoryDialog extends AddRepositoryDialog {
 		return new AddArtifactRepositoryOperation(ProvAdminUIMessages.AddArtifactRepositoryDialog_OperationLabel, url);
 	}
 
-	protected DefaultURLValidator createURLValidator() {
-		return new DefaultURLValidator() {
+	protected URLValidator createURLValidator() {
+		return new URLValidator() {
 			protected IStatus validateRepositoryURL(URL location, boolean contactRepositories, IProgressMonitor monitor) {
 				IStatus duplicateStatus = Status.OK_STATUS;
 				URL[] knownRepositories;
diff --git a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/AddColocatedRepositoryDialog.java b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/AddColocatedRepositoryDialog.java
index 1459841..481178c 100644
--- a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/AddColocatedRepositoryDialog.java
+++ b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/AddColocatedRepositoryDialog.java
@@ -16,7 +16,7 @@ import org.eclipse.equinox.internal.provisional.p2.core.ProvisionException;
 import org.eclipse.equinox.internal.provisional.p2.engine.IProfile;
 import org.eclipse.equinox.internal.provisional.p2.ui.ProvUI;
 import org.eclipse.equinox.internal.provisional.p2.ui.dialogs.AddRepositoryDialog;
-import org.eclipse.equinox.internal.provisional.p2.ui.dialogs.DefaultURLValidator;
+import org.eclipse.equinox.internal.provisional.p2.ui.dialogs.URLValidator;
 import org.eclipse.equinox.internal.provisional.p2.ui.operations.*;
 import org.eclipse.swt.widgets.Shell;
 import org.eclipse.ui.statushandlers.StatusManager;
@@ -40,8 +40,9 @@ public class AddColocatedRepositoryDialog extends AddRepositoryDialog {
 		return new AddColocatedRepositoryOperation(getShell().getText(), url);
 	}
 
-	protected DefaultURLValidator createURLValidator() {
+	protected URLValidator createURLValidator() {
 		MetadataGeneratingURLValidator validator = new MetadataGeneratingURLValidator();
+		validator.setKnownRepositoriesFlag(repoFlag);
 		validator.setProfile(getProfile());
 		validator.setShell(getShell());
 		return validator;
diff --git a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/externalFiles/MetadataGeneratingURLValidator.java b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/externalFiles/MetadataGeneratingURLValidator.java
index 4f71115..94b5787 100644
--- a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/externalFiles/MetadataGeneratingURLValidator.java
+++ b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/externalFiles/MetadataGeneratingURLValidator.java
@@ -17,7 +17,7 @@ import org.eclipse.core.runtime.*;
 import org.eclipse.equinox.internal.provisional.p2.core.ProvisionException;
 import org.eclipse.equinox.internal.provisional.p2.engine.IProfile;
 import org.eclipse.equinox.internal.provisional.p2.engine.IProfileRegistry;
-import org.eclipse.equinox.internal.provisional.p2.ui.dialogs.DefaultURLValidator;
+import org.eclipse.equinox.internal.provisional.p2.ui.dialogs.DefaultMetadataURLValidator;
 import org.eclipse.equinox.internal.provisional.p2.ui.dialogs.URLValidator;
 import org.eclipse.equinox.internal.provisional.p2.ui.operations.ProvisioningUtil;
 import org.eclipse.swt.widgets.Shell;
@@ -27,7 +27,7 @@ import org.eclipse.ui.PlatformUI;
  * @since 3.4
  *
  */
-public class MetadataGeneratingURLValidator extends DefaultURLValidator {
+public class MetadataGeneratingURLValidator extends DefaultMetadataURLValidator {
 
 	Shell shell;
 	IProfile profile;
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/AddRepositoryDialog.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/AddRepositoryDialog.java
index 1f0f324..35e21ac 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/AddRepositoryDialog.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/AddRepositoryDialog.java
@@ -41,16 +41,17 @@ public abstract class AddRepositoryDialog extends StatusDialog {
 
 	Button okButton;
 	Text url;
-	DefaultURLValidator urlValidator;
+	URLValidator urlValidator;
 	static final String[] ARCHIVE_EXTENSIONS = new String[] {"*.jar;*.zip"}; //$NON-NLS-1$ 
 	static String lastLocalLocation = null;
 	static String lastArchiveLocation = null;
+	protected int repoFlag;
 
 	public AddRepositoryDialog(Shell parentShell, int repoFlag) {
 
 		super(parentShell);
+		this.repoFlag = repoFlag;
 		urlValidator = createURLValidator();
-		urlValidator.setKnownRepositoriesFlag(repoFlag);
 		setTitle(ProvUIMessages.AddRepositoryDialog_Title);
 	}
 
@@ -137,8 +138,10 @@ public abstract class AddRepositoryDialog extends StatusDialog {
 		return comp;
 	}
 
-	protected DefaultURLValidator createURLValidator() {
-		return new DefaultURLValidator();
+	protected URLValidator createURLValidator() {
+		DefaultMetadataURLValidator validator = new DefaultMetadataURLValidator();
+		validator.setKnownRepositoriesFlag(repoFlag);
+		return validator;
 	}
 
 	protected URLValidator getURLValidator() {
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/DefaultMetadataURLValidator.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/DefaultMetadataURLValidator.java
new file mode 100644
index 0000000..19a500c
--- /dev/null
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/DefaultMetadataURLValidator.java
@@ -0,0 +1,62 @@
+/*******************************************************************************
+ * Copyright (c) 2008 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ ******************************************************************************/
+
+package org.eclipse.equinox.internal.provisional.p2.ui.dialogs;
+
+import java.net.URL;
+import org.eclipse.core.runtime.*;
+import org.eclipse.equinox.internal.p2.ui.ProvUIActivator;
+import org.eclipse.equinox.internal.p2.ui.ProvUIMessages;
+import org.eclipse.equinox.internal.provisional.p2.core.ProvisionException;
+import org.eclipse.equinox.internal.provisional.p2.metadata.repository.IMetadataRepositoryManager;
+import org.eclipse.equinox.internal.provisional.p2.ui.operations.ProvisioningUtil;
+
+/**
+ * @since 3.4
+ *
+ */
+public class DefaultMetadataURLValidator extends URLValidator {
+
+	protected int repoFlag;
+
+	public DefaultMetadataURLValidator() {
+		repoFlag = IMetadataRepositoryManager.REPOSITORIES_ALL;
+	}
+
+	public void setKnownRepositoriesFlag(int flag) {
+		repoFlag = flag;
+	}
+
+	/* (non-Javadoc)
+	 * @see org.eclipse.equinox.internal.provisional.p2.ui.dialogs.URLValidator#validateRepositoryURL(boolean)
+	 */
+	protected IStatus validateRepositoryURL(URL location, boolean contactRepositories, IProgressMonitor monitor) {
+		IStatus duplicateStatus = Status.OK_STATUS;
+		URL[] knownRepositories;
+		try {
+			knownRepositories = ProvisioningUtil.getMetadataRepositories(repoFlag);
+		} catch (ProvisionException e) {
+			knownRepositories = new URL[0];
+		}
+		for (int i = 0; i < knownRepositories.length; i++) {
+			if (knownRepositories[i].toExternalForm().equalsIgnoreCase(location.toExternalForm())) {
+				duplicateStatus = new Status(IStatus.ERROR, ProvUIActivator.PLUGIN_ID, LOCAL_VALIDATION_ERROR, ProvUIMessages.AddRepositoryDialog_DuplicateURL, null);
+				break;
+			}
+		}
+		if (!duplicateStatus.isOK())
+			return duplicateStatus;
+
+		if (contactRepositories)
+			return ProvisioningUtil.validateMetadataRepositoryLocation(location, monitor);
+		return duplicateStatus;
+	}
+}
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/DefaultURLValidator.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/DefaultURLValidator.java
deleted file mode 100644
index 42dcda6..0000000
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/DefaultURLValidator.java
+++ /dev/null
@@ -1,62 +0,0 @@
-/*******************************************************************************
- * Copyright (c) 2008 IBM Corporation and others.
- * All rights reserved. This program and the accompanying materials
- * are made available under the terms of the Eclipse Public License v1.0
- * which accompanies this distribution, and is available at
- * http://www.eclipse.org/legal/epl-v10.html
- *
- * Contributors:
- *     IBM Corporation - initial API and implementation
- ******************************************************************************/
-
-package org.eclipse.equinox.internal.provisional.p2.ui.dialogs;
-
-import java.net.URL;
-import org.eclipse.core.runtime.*;
-import org.eclipse.equinox.internal.p2.ui.ProvUIActivator;
-import org.eclipse.equinox.internal.p2.ui.ProvUIMessages;
-import org.eclipse.equinox.internal.provisional.p2.core.ProvisionException;
-import org.eclipse.equinox.internal.provisional.p2.metadata.repository.IMetadataRepositoryManager;
-import org.eclipse.equinox.internal.provisional.p2.ui.operations.ProvisioningUtil;
-
-/**
- * @since 3.4
- *
- */
-public class DefaultURLValidator extends URLValidator {
-
-	protected int repoFlag;
-
-	public DefaultURLValidator() {
-		repoFlag = IMetadataRepositoryManager.REPOSITORIES_ALL;
-	}
-
-	public void setKnownRepositoriesFlag(int flag) {
-		repoFlag = flag;
-	}
-
-	/* (non-Javadoc)
-	 * @see org.eclipse.equinox.internal.provisional.p2.ui.dialogs.URLValidator#validateRepositoryURL(boolean)
-	 */
-	protected IStatus validateRepositoryURL(URL location, boolean contactRepositories, IProgressMonitor monitor) {
-		IStatus duplicateStatus = Status.OK_STATUS;
-		URL[] knownRepositories;
-		try {
-			knownRepositories = ProvisioningUtil.getMetadataRepositories(repoFlag);
-		} catch (ProvisionException e) {
-			knownRepositories = new URL[0];
-		}
-		for (int i = 0; i < knownRepositories.length; i++) {
-			if (knownRepositories[i].toExternalForm().equalsIgnoreCase(location.toExternalForm())) {
-				duplicateStatus = new Status(IStatus.ERROR, ProvUIActivator.PLUGIN_ID, LOCAL_VALIDATION_ERROR, ProvUIMessages.AddRepositoryDialog_DuplicateURL, null);
-				break;
-			}
-		}
-		if (!duplicateStatus.isOK())
-			return duplicateStatus;
-
-		if (contactRepositories)
-			return ProvisioningUtil.validateMetadataRepositoryLocation(location, monitor);
-		return duplicateStatus;
-	}
-}
